src/main/kotlin/org/flaxo/plagiarism/data2graph.kt

```diff
@@ -119,15 +119,21 @@ fun Viz.refreshGraph(nodes: List<GraphNode>, links: List<GraphLink>, simulation:
     val threshold = inputById("plagiarismMatchThreshold").value.toInt()
     spanById("plagiarismMatchThresholdMonitor").innerHTML = threshold.toString()
     val directionEnabled = inputById("graphDirectionEnabledInput").checked
+    val showAllNodes = !inputById("hideExtraNodesEnabledInput").checked
 
     val dots = getDots(nodes)
     val arrows = getArrows(links)
 
+    val visibleNames = findVisibleNames(arrows, threshold)
+
     // Updating dots according to the simulation state.
     dots.forEachIndexed { index, dot ->
         val forceNode = simulation.nodes[index]
         dot.update(forceNode)
-        dot.show()
+
+        if (showAllNodes || dot.isVisible(visibleNames))
```

Comment:
Probably the `if` expression can be formatted to a single line rather then three lines.

Reply:
Inlined it